Code/Editor/EditorFramework/Assets/AssetTableWriter.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <EditorFramework/EditorFrameworkDLL.h>
+
+#include <EditorFramework/Assets/AssetCurator.h>
+#include <Foundation/Application/Config/FileSystemConfig.h>
+#include <Foundation/Configuration/Singleton.h>
+#include <Foundation/Threading/TaskSystem.h>
+
+
+struct ezAssetCuratorEvent;
+class ezTask;
+struct ezAssetInfo;
+
+/// \brief Asset table class. Persistent cache for an asset table.
+///
+/// The following assumptions need to be true for this cache to work:
+/// 1. ezAssetDocumentManager::AddEntriesToAssetTable does never change over time
+/// 2. ezAssetDocumentManager::GetAssetTableEntry never changes over the lifetime of an asset.
```

Comment:
For each sub-asset we only write the main output (the one without any tags) into the asset table and the assumption is, that this output always points to the same path. In the case of the material, it will always be the ezMaterialBin regardless of how the material is generated.